src/transform/src/normalize_lets.rs

```diff
@@ -254,33 +254,20 @@ mod support {
     ///
     /// An equivalent expression can be reconstructed by introducing the bindings atop the resulting `expr`.
     pub(super) fn digest_lets(expr: &mut MirRelationExpr) -> BTreeMap<LocalId, MirRelationExpr> {
-        let mut lets = BTreeMap::new();
-        let mut worklist = Vec::new();
-        digest_lets_helper(expr, &mut worklist);
-        while let Some((id, mut expr)) = worklist.pop() {
-            digest_lets_helper(&mut expr, &mut worklist);
-            lets.insert(id, expr);
-        }
-        lets
-    }
+        use mz_expr::visit::Visit;
 
-    /// Extract all `Let` bindings from `expr`, into `(id, value)` pairs.
-    ///
-    /// Importantly, the `value` pairs may not be `Let`-free, and must be further processed.
-    fn digest_lets_helper(
-        expr: &mut MirRelationExpr,
-        bindings: &mut Vec<(LocalId, MirRelationExpr)>,
-    ) {
-        let mut to_visit = vec![expr];
-        while let Some(expr) = to_visit.pop() {
+        let mut lets = BTreeMap::new();
+        #[allow(deprecated)]
```

Comment:
Since `action()` returns a `TransformError`, couldn't we use the non `_nolimit` call here?  There are no other users of `digest_lets`.

Reply:
We can, I'm not sure that we should.

This is a draft PR as I wanted buildkite to check for me whether the alternative `digest_lets` version that I had in mind was correct after [this discussion](https://github.com/MaterializeInc/materialize/pull/16387#discussion_r1040257464).

I think the first step is to see if we can / should un-deprecate the `_nolimit` functions using the `MirRelationExpr` implementations that we have at the moment.